app/services/query-builder.js

```diff
@@ -76,9 +76,18 @@ export const associationsQuery = ({ index, page, params, size }) => {
     } else if (params.status === 'Verlopen') {
       addFilter(
         ':query:recognitions.validityPeriod',
-        `(NOT ((recognitions.validityPeriod.startTime:<=${today}) AND (recognitions.validityPeriod.endTime:>=${today})))`,
+        `(NOT ((recognitions.validityPeriod.startTime:<=${today}) AND (recognitions.validityPeriod.endTime:>=${today}))) AND (recognitions.validityPeriod.endTime:<${today})`,
       );
       addFilter(':has:recognitions.validityPeriod.endTime', true);
+    } else if (
+      params.status === 'Erkend,Verlopen' ||
+      params.status === 'Verlopen,Erkend'
+    ) {
+      // Filter out "upcoming" recognitions when both filters are selected
+      addFilter(
+        ':query:recognitions.validityPeriod',
+        `(NOT ((recognitions.validityPeriod.startTime:>${today}) AND (recognitions.validityPeriod.endTime:>${today})))`,
+      );
```

Comment:
Hmm, this seemed to work at first, but now it also filters out organizations that have both an active recognition and a future one so this isn't correct.

Reply:
Not sure we can make this work with future recognitions.. We can not really filter them out I think..

Reply:
This is the closest I get: 
`  if (params.status === 'Erkend') {
      addFilter(
        ':query:recognitions.validityPeriod',
        `((recognitions.validityPeriod.startTime:<=${today}) AND (recognitions.validityPeriod.endTime:>=${today})) OR (NOT (recognitions.validityPeriod.startTime:>${today}) AND  (recognitions.validityPeriod.endTime:>${today}))`,
      );
      addFilter(':has:recognitions.validityPeriod.endTime', true);
    } else if (params.status === 'Verlopen') {
      addFilter(
        ':query:recognitions.validityPeriod',
        `(NOT ((recognitions.validityPeriod.startTime:<=${today}) AND (recognitions.validityPeriod.endTime:>=${today}))) AND NOT (recognitions.validityPeriod.startTime:>${today})  `,
      );
      addFilter(':has:recognitions.validityPeriod.endTime', true);
    } else if (
      params.status === 'Erkend,Verlopen' ||
      params.status === 'Verlopen,Erkend'
    ) {
      // Filter out "upcoming" recognitions when both filters are selected
      addFilter(
        ':query:recognitions.validityPeriod',
        `((recognitions.validityPeriod.startTime:<=${today}) AND (recognitions.validityPeriod.endTime:>=${today})) OR ((recognitions.validityPeriod.startTime:<=${today}) AND (recognitions.validityPeriod.endTime:<=${today}))`,
      );
    }`

Reply:
@Andresdev02 So I just take your version and we consider it "done" for now? Or could you commit the change if you already have it running locally?

Reply:
@Windvis yes we can do that for now. We need to check this again with Sofie to see where we can improve this. 
For me you can take the above version. I don't have it locally anymore or let me know if you want me to do it. 🙂

Reply:
@Andresdev02 I've updated it to use your version :+1: